src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1406,10 +1409,18 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             DominantListStructure domintListStructure,
             StackFrame stackFrame, RuntimeCore runtimeCore)
         {
-            // Update the CallsiteExecutionState with 
-            // TODO: Replace this with the real data
-            UpdateCallsiteExecutionState(null, runtimeCore);
+           
+            // if the last dispatched callsite is this callsite then we are repeatedly making calls
+            // to this same callsite (for example replicating over an outer function that contains this callsite)
+            // and should not reset the invoke count.
+            if (runtimeCore.LastDispatchedCallSite != this)
```

Comment:
Did this not work:
```
if(this.id != runtimeCore.DSExecutable.ExecutingGraphnode.CallsiteIdentifier) {...}
```

Reply:
yes, as stated above, I found this did not work - the `ExecutingGraphNode` on the executable has already been set to the current callsite - so it's too late, we really need the `LastGraphNode` - not the current one.